apps/src/templates/lessonOverview/activities/LevelDetailsDialog.jsx

```diff
@@ -64,9 +97,53 @@ export default class LevelDetailsDialog extends Component {
     }
   }
 
+  handleBubbleChoiceBubbleClick = level => {
+    const previousSelected = this.state.selectedLevel;
+    const clonedScriptLevel = _.cloneDeep(this.state.scriptLevel);
+    if (previousSelected.name === clonedScriptLevel.level.name) {
+      clonedScriptLevel.highlighted = false;
+    } else {
+      const clonedPreviousSelected = clonedScriptLevel.sublevels.find(
+        sublevel => sublevel.name === previousSelected.name
+      );
+      clonedPreviousSelected.highlighted = false;
+    }
+    if (level.level) {
```

Comment:
this is kinda confusing. Is this really ScriptLevel.level?

I'm finding this method tough to fully understand and would love to some comments to help explain what it is doing at each step.

Reply:
I think its hard to keep track of what is the parent level, the child(ren) levels and what is the script level that contains the parent level.

Reply:
Yeah the problem is that on the bubble click, sometimes the script level is passed through (if the big bubble is clicked) and sometimes the level is passed through (if any of the sublevels are clicked).

Added some comments and changed a variable name to hopefully make it a bit clearer. Let me know if there's anything else I can fix here.